### ISSUE TITLE
Plugins report level change doesn't work

### ISSUE THREAD

**Issue by AlexWayfer** [remarkjs/remark-lint#265]
### Subject of the issue

I want to change report level of some plugins from `warn` to `error` in YAML config.

### Your environment

*   **OS**: Arch Linux.
*   **Packages**:
    ```json
    "remark": "^13.0.0",
    "remark-gfm": "^1.0.0",
    "remark-preset-lint-markdown-style-guide": "^4.0.0",
    ```
*   **Env**: Node.js 15.8.0, Yarn 1.22.10

### Steps to reproduce

I have a config like this:

```yaml
# .remarkrc.yaml
plugins:
  - - lint-no-shell-dollars
    - error
```

### Expected behavior

Now `lint-no-shell-dollars` reports as errors.

### Actual behavior

It reports as `warn` further.

Also I can't off it via `off`, only via `false`, regardless to [documentation](https://github.com/remarkjs/remark-lint/blob/b40fe0f/doc/rules.md#configuration).

The same with `lint-list-item-indent`, for example.

Nested arrays like:

```yaml
  - - lint-no-shell-dollars
    - - off
```

don't work too:

```
README.md
  1:1  error  TypeError: value.concat is not a function
    at coerce (/home/alex/Projects/rubyapi/node_modules/unified-lint-rule/index.js:80:20)
    at Function.remark-lint:no-shell-dollars (/home/alex/Projects/rubyapi/node_modules/unified-lint-rule/index.js:24:18)
    at freeze (/home/alex/.config/yarn/global/node_modules/remark/node_modules/unified/index.js:123:28)
    at Function.parse (/home/alex/.config/yarn/global/node_modules/remark/node_modules/unified/index.js:272:5)
    at parse (/home/alex/.config/yarn/global/node_modules/unified-args/node_modules/unified-engine/lib/file-pipeline/parse.js:41:36)
    at wrapped (/home/alex/.config/yarn/global/node_modules/trough/wrap.js:25:19)
    at next (/home/alex/.config/yarn/global/node_modules/trough/index.js:58:24)
    at done (/home/alex/.config/yarn/global/node_modules/trough/wrap.js:56:16)
    at handleConfiguration (/home/alex/.config/yarn/global/node_modules/unified-args/node_modules/unified-engine/lib/file-pipeline/configure.js:68:5)
    at done (/home/alex/.config/yarn/global/node_modules/unified-args/node_modules/unified-engine/lib/configuration.js:76:14)
```

**Comment by wooorm** [comment]
I do know that yaml sees an unquoted "off" as false. That might (helpt to) explain *some* of what's going on here
Otherwise: try the object form (key: value) instead of arrays?

**Comment by AlexWayfer** [comment]
> I do know that yaml sees an unquoted "off" as false. That might (helpt to) explain _some_ of what's going on here

That should make `off` works as well as `false`, but the irony that `false` works and `off` doesn't.



> Otherwise: try the object form (key: value) instead of arrays?

I don't see such form in the mentioned documentation, and my tries like these do nothing (even with `false`):

```yaml
  - lint-no-file-name-irregular-characters: false
  # or
  - - lint-no-file-name-irregular-characters: false
```

**Comment by wooorm** [comment]
I did get:

```yml
plugins:
  - preset-wooorm
  - - lint-no-html
    - false
```

to work.

For strings or numbers, you have to pass them as an array. That’s shown here: https://github.com/remarkjs/remark-lint/blob/main/doc/rules.md.

E.g.:

```yml
plugins:
  - preset-wooorm
  - - lint-no-html
    - - off
```

Or:

```yml
plugins:
  - preset-wooorm
  - - lint-no-html
    - - 0
```

**Comment by wooorm** [comment]
The reason for this is that it’s preferred to configure plugins. Not to change their severity, so:

```yml
plugins:
  - preset-wooorm
  - - maximum-line-length
    - 70
```

Would set the line length to 70. Not some very high severity level!

**Comment by wooorm** [comment]
For the object form I meant:

```yml
plugins:
  preset-wooorm: true
  lint-no-html: false
```

**Comment by AlexWayfer** [comment]
@wooorm did you see the whole issue's description? As I wrote, syntax with nested arrays causes errors `TypeError: value.concat is not a function`.

**Comment by wooorm** [comment]
These are all examples I just tried and they worked fine?

**Comment by AlexWayfer** [comment]
> These are all examples I just tried and they worked fine?

I don't know or don't understand.

Here is an example repository (yeah, I didn't want to create it) even with CI: https://github.com/AlexWayfer/remark_yaml_test

---

And… via the second commit I found the problem: you should to add `remark-cli` as dependency.

I don't know how it worked without it, maybe it's a `nodenv` problem, or `package.json`'s scripts, but [CI displayed this local-related problem](https://cirrus-ci.com/task/6432121456689152).

Sorry, I'll try to remember.

**Comment by wooorm** [comment]
Maybe you have an ancient `remark-cli` installed globally? 🤔 I don’t think it’s used in many other packages so I don’t see it coming from another place in your local `node_modules/`?

**Comment by wooorm** [comment]
Could you do `remark -v`? What does that give you locally?

**Comment by AlexWayfer** [comment]
> Maybe you have an ancient `remark-cli` installed globally?  I don’t think it’s used in many other packages so I don’t see it coming from another place in your local `node_modules/`?

Yep, I had, and my debug was:

```
$ remark -v
remark: 9.0.0, remark-cli: 5.0.0

$ npm run remark

> remark
> remark -v

remark: 13.0.0, remark-cli: 9.0.0
```

`yarn global remove remark-cli` should prevent such errors in the future, I hope.

**Comment by wooorm** [comment]
`remarkp-cli@5.0.0`! that’s awhile! Well, glad this works now!

**Comment by AlexWayfer** [comment]
> `remarkp-cli@5.0.0`! that’s awhile! Well, glad this works now!

([asking `type/question` since 2018](https://github.com/remarkjs/remark-lint/issues?q=author%3AAlexWayfer+sort%3Acreated-asc) 😅  really ashamed sometimes)

**Comment by wooorm** [comment]
Haha nooo don't be! That's fine! 💪
